client/src/pages/ChatPage.js

```diff
@@ -0,0 +1,315 @@
+} from "../store/authentication/thunks";
+import { useNavigate } from "react-router-dom";
+
+// Credits: Setting up socket io for chat
+// https://dev.to/bhavik786/building-a-real-time-chat-application-using-mern-stack-and-socketio-1obn
+const ChatPage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const chat = useSelector((state) => state.chatReducer.chat);
+  const user = useSelector((state) => state.loginReducer);
+  const groupID = window.location.pathname.split("/").pop();
+
+  const [socket, setSocket] = useState(null);
+  const [message, setMessage] = useState("");
+  const [openUser, setOpenUser] = useState(false);
+  const [openName, setOpenName] = useState(false);
+  const [openExit, setOpenExit] = useState(false);
+
+  const [name, setName] = useState(user.user ? user.user.groupID : "");
+  const [userInvite, setUserInvite] = useState(
+    user.user ? user.user.groupID : ""
+  );
+  const username = user.user ? user.user.username : "";
+
+  useEffect(() => {
+    dispatch(getChatHistoryAsync(groupID));
+  }, [dispatch, groupID]);
+
+  useEffect(() => {
+    const newSocket = io("http://localhost:8080");
+    setSocket(newSocket);
+
+    return () => {
+      newSocket.disconnect();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (socket) {
+      socket.on("message", handleMessage);
+    }
+
+    return () => {
+      if (socket) {
+        socket.off("message", handleMessage);
+      }
+    };
+  }, [socket]);
+
+  const handleMessage = (message) => {
+    dispatch(putChatHistoryAsync({ groupID, newMessage: message }));
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (message) {
+      socket.emit("sendMessage", { username, message });
+      setMessage("");
+    }
+  };
+
+  const addUserHandler = () => {
+    setOpenUser((cur) => !cur);
+  };
+
+  const exitUserHandler = () => {
+    setOpenExit((cur) => !cur);
+  };
+
+  const changeNameHandler = () => {
+    setOpenName((cur) => !cur);
+  };
+
+  const onChangeName = (event) => {
+    dispatch(
+      renameChatAsync({
+        groupID: groupID,
+        name: name,
+      })
+    );
+    changeNameHandler();
+  };
+
+  const onInviteUser = (event) => {
+    setUserInvite(event.target.value);
+  };
+
+  const onExitUser = () => {
+    dispatch(
+      leaveChatAsync({
+        username: username,
+        groupID: groupID,
+      })
+    );
+    exitUserHandler();
+    navigate(`/studyGroups`);
+  };
+
+  const inviteDispatch = () => {
+    const updatedUser = {
+      username: userInvite,
+      groupID: groupID,
+    };
+
+    dispatch(inviteUserAsync(updatedUser));
+
+    addUserHandler();
+  };
+
+  if (!user.user) {
+    return null;
+  }
+
+  return (
+    <div>
+      <div className="fixed left-0 top-0 z-[1035] h-screen">
+        <SideBar />
+      </div>
+      <div className="p-5 !pl-[300px]">
+        <div className="flex justify-center">
+          <Chip
+            value={chat.name}
```

Comment:
Perhaps in the future we should also limit chat name length or introduce wrapping into the chip?:

![image](https://github.com/awctw/study-dash/assets/56493743/939f549f-3e20-4cdd-b134-8a27216e75a0)

Reply:
Yeah, will definitely limit it in the future, I've truncated the text for now tho so it won't overflow